src/Apps/Auction/__tests__/routes.test.ts

```diff
@@ -24,84 +28,208 @@ describe("Auction/routes", () => {
     })
   }
 
-  const mockResolver = (
+  const mockRegisterResolver = (
     data: DeFragedRegisterQueryResponse
   ): DeFragedRegisterQueryResponse => ({
     sale: data.sale,
     me: data.me,
   })
 
+  const mockConfirmBidResolver = ({
+    artwork = {},
+    saleArtwork = {},
+    sale = {},
+    me = {},
+  }: Partial<{
+    artwork: Partial<BidQueryResponse["artwork"]>
+    sale: Partial<BidQueryResponse["artwork"]["saleArtwork"]["sale"]>
+    saleArtwork: Partial<BidQueryResponse["artwork"]["saleArtwork"]>
+    me: Partial<BidQueryResponse["me"]>
+  }> = {}): BidQueryResponse => ({
+    ...BidQueryResponseFixture,
+    artwork: {
+      ...BidQueryResponseFixture.artwork,
+      ...artwork,
+      saleArtwork: {
+        ...BidQueryResponseFixture.artwork.saleArtwork,
+        ...saleArtwork,
+        sale: {
+          ...BidQueryResponseFixture.artwork.saleArtwork.sale,
+          ...sale,
+        },
+      },
+    },
+    me: {
+      ...BidQueryResponseFixture.me,
+      ...me,
+    },
+  })
+
   it("renders the Auction FAQ view", async () => {
-    const { status } = await render("/auction-faq", mockResolver(Fixture))
+    const { status } = await render("/auction-faq", {})
 
     expect(status).toBe(200)
   })
 
-  it("does not redirect if a sale is found", async () => {
-    const { redirect, status } = await render(
-      `/auction-registration/${Fixture.sale.id}`,
-      mockResolver(Fixture)
-    )
+  describe("Confirm Bid: /:saleId/bid/:artworkId", () => {
+    it("does not redirect if the user is qualified to bid in the sale, the sale is open, and the artwork is biddable", async () => {
+      const fixture: BidQueryResponse = mockConfirmBidResolver()
+      const { redirect, status } = await render(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/bid/${
+          fixture.artwork.id
+        }`,
+        fixture
+      )
 
-    expect(status).toBe(200)
-    expect(redirect).toBeUndefined
-  })
+      expect(status).toBe(200)
+      expect(redirect).toBeUndefined
+    })
 
-  it("also responds to auction-registration2 route", async () => {
-    const { status } = await render(
-      `/auction-registration2/${Fixture.sale.id}`,
-      mockResolver(Fixture)
-    )
+    it("redirects to confirm registration page if user is registered but not qualified to bid (to remind them)", async () => {
+      const fixture: BidQueryResponse = mockConfirmBidResolver({
+        sale: { registrationStatus: { qualified_for_bidding: false } },
+      })
+      const { redirect } = await render(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/bid/${
+          fixture.artwork.id
+        }`,
+        fixture
+      )
 
-    expect(status).toBe(200)
-  })
+      expect(redirect.url).toBe(
+        `/auction/${fixture.artwork.saleArtwork.sale.id}/confirm-registration`
```

Comment:
My assumption here is that this will re-trigger the 'registration pending' modal because the user has somehow tried to bid on an artwork when they should not have been able to.